cloud/deployment/workerqueue/workerqueue.go

```diff
@@ -35,264 +35,329 @@ var (
+		listToCreate                         []astroplatformcore.WorkerQueueRequest
+		existingQueues                       []astroplatformcore.WorkerQueue
+		hybridListToCreate                   []astroplatformcore.HybridWorkerQueueRequest
+		defaultOptions                       astroplatformcore.WorkerQueueOptions
 	)
 	// get or select the deployment
-	requestedDeployment, err = deployment.GetDeployment(ws, deploymentID, deploymentName, true, client, coreClient)
+	requestedDeployment, err = deployment.GetDeployment(ws, deploymentID, deploymentName, true, platformCoreClient, coreClient)
 	if err != nil {
 		return err
 	}
 
-	if requestedDeployment.ID == "" {
+	if requestedDeployment.Id == "" {
 		fmt.Printf("No Deployments found in workspace %s\n", ansi.Bold(ws))
 		return nil
 	}
 
-	if deployment.IsDeploymentHosted(requestedDeployment.Type) || deployment.IsDeploymentDedicated(requestedDeployment.Type) {
-		nodePoolID = requestedDeployment.Cluster.NodePools[0].ID
-		configOptions, err := client.GetDeploymentConfig()
+	if deployment.IsDeploymentStandard(*requestedDeployment.Type) || deployment.IsDeploymentDedicated(*requestedDeployment.Type) {
+		getDeploymentOptions := astroplatformcore.GetDeploymentOptionsParams{
+			DeploymentId: &requestedDeployment.Id,
+		}
+		deploymentOptions, err := deployment.GetPlatformDeploymentOptions("", getDeploymentOptions, platformCoreClient)
 		if err != nil {
 			return err
 		}
-		astroMachines := configOptions.AstroMachines
+		WorkerMachines := deploymentOptions.WorkerMachines
 		// get the machine to use
-		astroMachine, err = selectAstroMachine(workerType, astroMachines, out)
+		workerMachine, err = selectWorkerMachine(workerType, WorkerMachines, out)
 		if err != nil {
 			return err
 		}
 
 		if wQueueConcurrency == 0 && action == createAction {
-			wQueueConcurrency = astroMachine.ConcurrentTasks // This is set based on the machine type the user chooses if not explicitly passed by the user
+			wQueueConcurrency = int(workerMachine.Concurrency.Default) // This is set based on the machine type the user chooses if not explicitly passed by the user
+		}
+		queueToCreateOrUpdate = &astroplatformcore.WorkerQueueRequest{
+			Name:              name,
+			IsDefault:         false, // cannot create a default queue
+			AstroMachine:      astroplatformcore.WorkerQueueRequestAstroMachine(workerMachine.Name),
+			MinWorkerCount:    wQueueMin,         // use the value from the user input
+			MaxWorkerCount:    wQueueMax,         // use the value from the user input
+			WorkerConcurrency: wQueueConcurrency, // use the value from the user input
+		}
+		fmt.Println(requestedDeployment)
+		// create listToCreate
+		if requestedDeployment.WorkerQueues != nil {
+			fmt.Println(requestedDeployment)
+			fmt.Println(requestedDeployment.WorkerQueues)
+			for i := range *requestedDeployment.WorkerQueues {
+				queues := *requestedDeployment.WorkerQueues
```

Comment:
we should move this out of the for loop - its being reinstantiated every time when it never changes

Reply:
moved